open-grant-proposals/open-proposal-twitter-thread-rollup-backed-by-filecoin-and-nfts.md

```diff
@@ -0,0 +1,127 @@
+# Open Grant Proposal: `Twitter Thread Rollup backed by Filecoin and NFTs`
+
+**Name of Project:** NFTShapshot
+
+**Proposal Category:** `app-dev`, `docs`
+
+**Proposer:** `@jamesmorgan`
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** `Yes`
+
+# Project Description
+
+- Twitter thread rollup and timestamp bots are not a new thing, however what they currently do not utilise distributed, decentralised storage, such as the Filecoin network and then allow the creator to turn them into a tradeable NFTs (non-fungible tokens). 
+
```

Comment:
Hi @jamesmorgan,

Question - how are Twitter thread rollups unique? (to be turned into NFTs?)

Reply:
(Unique or rare - or is this because Twitter threads may not be available forever?)

Reply:
hi @eshon - thanks for comments/questions. 

So the idea is that we would use the thread rollup as a unique fingerprint so there would only be one NFT of it.

Reply:
@jamesmorgan - Will you be creating your own NFT registry for this on Ethereum?

Reply:
Yes, the plan would be to have a simple NFT on Ethereum (ERC-721 most likely) which allows the app to registry and create NFTs from each thread. We can look to leverage OpenSea for a marketplace and then can pull from their API or stand up a simple DB to index and list the tokens, we've done this before for other projects.